Add PowerShell Invoke-WebRequest test runner

JeremyRand opened the pull request:
Feedback on structure welcome; I wasn't quite sure where in the repo to put the `.ps1` test script.

JackOfMostTrades commented on test-suites/impltests/powershell.go:
```diff
@@ -0,0 +1,43 @@
+
+// Compatibility notes:
+// This runner requires Administrator privileges so that it can add/remove root CA's.
+// You must also have "test-suites/scripts/" in your PATH.
```
Can you embed the script and write it somewhere, passing the absolute path to powershell below? See `node.go` for an example of this, although you're welcome to use the [embed package](https://pkg.go.dev/embed) rather than inlining the script the way I did for Node.

JeremyRand replied:
Indeed, that does seem like a better approach.  I'll update this PR shortly.

JeremyRand replied:
PR updated.

JackOfMostTrades commented on test-suites/impltests/powershell.go:
```diff
@@ -0,0 +1,43 @@
+
+func (c *PowerShellRunner) Initialize() error {
+	var err error
+	c.version, err = execAndCapture("powershell", "$PSVersionTable.PSEdition + \" \" + $PSVersionTable.PSVersion")
```
I'm not sure if this already includes it, but are you able to capture the windows version (e.g. build number) here as well? My hunch is that under the hood powershell will be using a Windows-provided certificate verification stack rather than something necessarily powershell specific, so having the Windows version would probably be useful.

JeremyRand replied:
Yeah that probably would be helpful; I'll update this PR shortly.

JeremyRand replied:
PR updated.

JackOfMostTrades reviewed (changes requested):
Looks good to me, just a couple of suggestions. Have you tried running this? I can do so and add some archived results once it's merged, but I'm curious if any interesting results have popped up in the coarse of implementing the powershell runner. :)

JeremyRand commented:
I think I've addressed all feedback, please let me know if anything else needs tweaking.

> Have you tried running this? I can do so and add some archived results once it's merged, but I'm curious if any interesting results have popped up in the coarse of implementing the powershell runner. :)

Yes, I ran it locally; if you like, I can add the results file to the PR.  It picked up a few apparent bugs in PowerShell's handling of name constraints (and also found that PowerShell doesn't appear to blacklist insecure crypto), but I had difficulty analyzing the name constraints bugs due to incomplete documentation of BetterTLS (I intend to file some issues and/or PR's for those separately).  If my cursory analysis is correct, the PowerShell bugs in name constraints handling seem mostly related to either IP address constraints or Subject Common Name handling.  Once I've had time to analyze better, I will probably reach out to Microsoft about fixing bugs as needed.

JackOfMostTrades commented:
Cool, looks great, thanks!

> PowerShell bugs in name constraints handling seem mostly related to either IP address constraints or Subject Common Name handling

That sounds kind of familiar. Back in the day, Internet Explorer had an issue where you could effectively bypass for IP address name constraints by putting an otherwise disallowed IP in the subject common name. if you get any traction reporting it, I'd love to know. :)
